src/jackdaw/streams/interop.clj

```diff
@@ -314,6 +324,26 @@
      (.selectKey ^KStream kstream
                  ^KeyValueMapper (select-key-value-mapper select-key-value-mapper-fn))))
 
+  (join-global
+    [_ global-ktable key-value-mapper-fn joiner-fn]
+    (clj-kstream
+     (.join kstream
+            ^GlobalKTable (global-ktable* global-ktable)
+            ^KeyValueMapper (select-key-value-mapper key-value-mapper-fn)
+            ^ValueJoiner (value-joiner joiner-fn))))
+
+  (left-join-global
+    [_ global-ktable key-value-mapper-fn joiner-fn]
+    (clj-kstream
+     (.leftJoin kstream
+                ^GlobalKTable (global-ktable* global-ktable)
+                ^KeyValueMapper (select-key-value-mapper key-value-mapper-fn)
+                ^ValueJoiner (value-joiner joiner-fn))))
+
+  (kstream* [_]
+    kstream)
+
+  ITransformingKStream
```

Comment:
Added a new protocol for the transform fns on a stream, which is always used wherever IKStream is used. This had to be done to allow Coverage to run over this IKStream protocol.

Its not ideal, and perhaps there is a  newer / different code coverage tool to fix this.

The change is however invisible to users of Jackdaw.